apollo-router/src/plugins/record_replay/runner.rs

```diff
@@ -0,0 +1,46 @@
+#[cfg(test)]
+mod tests {
+    use console::style;
```

Comment:
This seems like it would be better named as `replay_tests.rs` and it should be a direct submodule or `replay.rs`. Also, do something like this to restructure things a little:
```
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        renamed:    apollo-router/src/plugins/record_replay/runner.rs -> apollo-router/src/plugins/record_replay/replay_tests.rs

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   apollo-router/src/plugins/record_replay/mod.rs
        modified:   apollo-router/src/plugins/record_replay/replay.rs
        modified:   apollo-router/src/plugins/record_replay/replay_tests.rs
```

The code changes would look like:
```
garypen@Garys-MacBook-Pro router % git diff -b
diff --git a/apollo-router/src/plugins/record_replay/mod.rs b/apollo-router/src/plugins/record_replay/mod.rs
index 893ec15fe..b3f42fffa 100644
--- a/apollo-router/src/plugins/record_replay/mod.rs
+++ b/apollo-router/src/plugins/record_replay/mod.rs
@@ -1,4 +1,3 @@
 mod record;
 mod recording;
 mod replay;
-mod runner;
diff --git a/apollo-router/src/plugins/record_replay/replay.rs b/apollo-router/src/plugins/record_replay/replay.rs
index d3234d21e..d4f7e39a8 100644
--- a/apollo-router/src/plugins/record_replay/replay.rs
+++ b/apollo-router/src/plugins/record_replay/replay.rs
@@ -412,3 +412,7 @@ fn render_invisible(s: &str, newlines_matter: bool) -> Cow<'_, str> {
         Cow::Borrowed(s)
     }
 }
+
+#[cfg(test)]
+#[path = "replay_tests.rs"]
+mod tests;
diff --git a/apollo-router/src/plugins/record_replay/replay_tests.rs b/apollo-router/src/plugins/record_replay/replay_tests.rs
index efc13011e..6b79aa379 100644
--- a/apollo-router/src/plugins/record_replay/replay_tests.rs
+++ b/apollo-router/src/plugins/record_replay/replay_tests.rs
@@ -1,13 +1,11 @@
-#[cfg(test)]
-mod tests {
-    use console::style;
-    use tower::ServiceExt;
+use console::style;
+use tower::ServiceExt;
 
-    use super::super::replay::Replay;
-    use crate::TestHarness;
+use super::Replay;
+use crate::TestHarness;
 
-    #[tokio::test]
-    async fn replay_recording() {
+#[tokio::test]
+async fn replay_recording() {
     let recording_file = if let Ok(file) = std::env::var("RECORDING_FILE") {
         file
     } else {
@@ -42,5 +40,4 @@ mod tests {
             println!();
         }
     }
-    }
 }
```

Reply:
i like it!